bottles/frontend/windows/launchoptions.py

```diff
@@ -339,6 +350,34 @@ def __reset_cwd(self, *_args):
         self.action_cwd.set_subtitle(self.__default_cwd_msg)
         self.btn_cwd_reset.set_visible(False)
 
+    def __choose_midi_soundfont(self, *_args):
+        def set_path(dialog, response):
+            if response != Gtk.ResponseType.ACCEPT:
+                self.action_midi_soundfont.set_subtitle(self.__default_midi_soundfont_msg)
+                return
+
+            directory_path = dialog.get_file().get_path()
+            self.program["midi_soundfont"] = directory_path
+            self.action_midi_soundfont.set_subtitle(directory_path)
+            self.btn_midi_soundfont_reset.set_visible(True)
+
+        dialog = Gtk.FileChooserNative.new(
```

Comment:
Since [`Gtk.FileChooserNative` is deprecated](https://docs.gtk.org/gtk4/class.FileChooserNative.html), it'd be better to use [`Gtk.FileDialog`](https://docs.gtk.org/gtk4/class.FileDialog.html).

You can take this as a reference: https://gitlab.gnome.org/World/Upscaler/-/commit/68fbab835820f2e60d007636c16db8e91dfe4fd3

Reply:
I actually thought about that, but ended up using `Gtk.FileChooserNative` as it needed filters applied, and the way to do it via `Gtk.FileDialog.set_filters` instead seemed more complicated. However, that snippet of yours is pretty direct on how to do it.

If we're going to go that way, though, we should probably adapt it for _all_ dialogs that still use `Gtk.FileChooserNative` (especially the ones that employ filters, as otherwise [filters.py](https://github.com/bottlesdevs/Bottles/blob/0c07176401736c10e9c70f19d82ff231eab56969/bottles/frontend/utils/filters.py) will look like a mess).

Reply:
@TheEvilSkeleton This should do the trick (in a transitional way):
```python
def __set_filter(dialog: GObject.Object, title: str, patterns: tuple):
    filter = Gtk.FileFilter()
    filter.set_name(title)
    for pattern in patterns:
        filter.add_pattern(pattern)
    
    if isinstance(dialog, Gtk.FileDialog):
        filters = Gio.ListStore.new(Gtk.FileFilter)
        filters.append(filter)
        dialog.set_filters(filters)
    elif isinstance(dialog, Gtk.FileChooserNative):
        dialog.add_filter(filter)
    else:
        raise TypeError
```